src/terminal/parser/stateMachine.cpp

```diff
@@ -1120,7 +1188,7 @@ void StateMachine::_EventEscapeIntermediate(const wchar_t wch)
 //   1. Execute C0 control characters
 //   2. Ignore Delete characters
 //   3. Collect Intermediate characters
-//   4. Begin to ignore all remaining parameters when an invalid character is detected (CsiIgnore)
+//   4. Begin to ignore all remaining parameters when a sub parameter character is detected (CsiIgnore)

```

Comment:
this comment seems wrong? Er, I guess `\x1b[:m` is invalid - is it?

Reply:
(might just need clarification)

Reply:
I think, I tried to keep the existing behavior here. Should we allow such sequences? I'd be happy make the changes.

Sub parameters to empty parameter shouldn't have any effect, AFAIK.

Reply:
> I think, I tried to keep the existing behavior here. Should we allow such sequences?

@zadjii-msft Should I make any changes here?

Reply:
In the specific context of `SGR`, the default parameter value is 0, so I'd consider `:` to be a sub parameter of 0 (essentially the equivalent of `0:0`). Since that doesn't have a defined meaning that we're aware of, it classifies as an unimplemented selective parameter, which should be ignored (per STD 070 §3.5.1.3).

So I'd say `printf "\e[:;4mhello\n"` _should_ render as underlined. The first parameter is ignored, but the second is still executed.

I wouldn't take much notice of what other terminals are doing unless we can find some spec text that backs up their interpretation, because I don't think they've put much thought into this other than doing whatever was easiest for them. I've tested about a dozen of them, and _none_ of them interpret sub parameters in exactly the same way.

Reply:
> So I'd say printf "\e[:;4mhello\n" should render as underlined. The first parameter is ignored, but the second is still executed.

I think this can easily be done now, since we have `CsiParamIgnore` to ignore one parameter at a time, or ignore everything up to ";" and enter `CsiParam` afterwards.

Reply:
After the discussion with @j4james, We let the parser to parse `\x1b[:m` without entering into `CsiIgnore`.. It is interpreted as `\x1b[<empty-parameter>:<empty-subparameter>m`, which makes it a no-op.